BeeSwift/JSONGoal.swift

```diff
@@ -614,11 +614,12 @@ class JSONGoal {
     func updateBeeminderWithValue(datapointValue : Double, daystamp : String, success: (() -> ())?, errorCompletion: (() -> ())?) {
         
         if datapointValue == 0  { return }
+        guard let username = CurrentUserManager.sharedManager.username else { return }
```

Comment:
If there is no username, does that mean we are logged out? If we are logged out, shall we then just cancel pending queries and background deliveries? Would that be something to do here in the else block? Or rather call explicitly on goals like is done for setting up HK integration?

Reply:
How about we create a bug ticket for deregistering from healthkit - on sign out or when goals change such that a previous query is no longer necessary - and allow the guard here as a fix

Reply:
created the new bug ticket as #146